### PR TITLE
Add Role routes and tests

### PR DESCRIPTION
**Pull request by tonymontaro**
#### What does this PR do?

This PR create the Role model, routes and adds tests for them

#### Description of Task to be completed?

- Integrate django rest framework
- Create tests for the Role model and routes/views
- Build Role model
- Build role views
- Authenticate and authorize routes

#### How should this be manually tested?

Visit the deployed app: https://montarojed-pydms.herokuapp.com/roles/
1. List Roles: https://montarojed-pydms.herokuapp.com/roles/
2. Create new role (must be admin): https://montarojed-pydms.herokuapp.com/roles/new
3. Get/Update/Delete a role (must be admin): https://montarojed-pydms.herokuapp.com/roles/<role_id>

### CODE REVIEW

File: main/tests.py [hunk 1]
```diff
@@ -1,3 +1,84 @@
...
+    def setUp(self):
+        user = User.objects.create(username='nerd')
+        self.role_name = 'editor'
+        self.role = Role(name=self.role_name)
+
+    def test_model_can_create_a_role(self):
+        old_count = Role.objects.count()
+        self.role.save()
+        new_count = Role.objects.count()
+        self.assertEqual(new_count, old_count + 1)
```
Comment (jomadoye) [inline main/tests.py:26]: Awesome PR @andela-angene is there a reason why you are only asserting that the number of roles have increased by +1, and not checking the name of the new role that was created.

File: main/tests.py [hunk 2]
```diff
@@ -1,3 +1,84 @@
...
+
+    def test_api_can_delete_role(self):
+        """Api can delete role"""
+        role = Role.objects.get()
+        response = self.client.delete(
+            reverse('role_details', kwargs={'pk': role.id}),
+            format='json',
+            follow=True
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```
Comment (jomadoye) [inline main/tests.py:85]: Thanks for observing TDD

Comment (jomadoye) [reply main/tests.py:85]: @andela-angene Please can you squash your commits into 1

File: main/tests.py [hunk 3]
```diff
@@ -1,3 +1,84 @@
...
+    def test_api_can_get_a_role(self):
+        """Api can get a role"""
+        role = Role.objects.get(id=1)
+        response = self.client.get(
+            '/roles/',
+            kwargs={'pk': role.id},
+            format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, role)
```
Comment (jomadoye) [inline main/tests.py:64]: @andela-angene Please can we assert that an exact role was gotten here, as opposed to checking the status code and if a role was gotten.

File: main/tests.py [hunk 4]
```diff
@@ -1,3 +1,84 @@
...
+
+    def test_api_can_update_role(self):
+        """"Api can update role"""
+        role = Role.objects.get(id=1)
+        res = self.client.put(
+            reverse('role_details', kwargs={'pk': role.id}),
+            {'name': 'Something_new'},
+            format='json'
+        )
```
Comment (jomadoye) [inline main/tests.py:73]: @andela-angene Please, can we assert the role was indeed updated, not only the status code.

File: main/tests.py (hunk 2 above)
Comment (tonymontaro) [reply main/tests.py:85]: The purpose of this project is to learn web development with django, the commits will be helpful here for review, squashing them defeats that purpose.

File: main/tests.py (hunk 4 above)
Comment (tonymontaro) [reply main/tests.py:73]: Good observation, check line 75.

**Pull request merged**